### DISCUSSION TITLE
Choosing which heading to add a capture only if a certain one doesn't exist

### DISCUSSION THREAD

**Discussion by asmagen** [chhoumann/quickadd#1292] [Q&A]
I'm trying to capture to any file I choose in the DB, to a default column called To Do, but in files where I have a different format for whatever reason, I want to be asked for which existing column to add to (but not asked when To Do exists). How can I do that?

Thanks

**Comment by chhoumann** [comment]
Coming back to this one very, very late - sorry about that. The good news: both halves of what you asked for exist now, and neither did back in 2022.

Option A, which I'd recommend: set **Insert after** to `## To Do` and enable **Create line if not found**. Files that already have the heading get the capture under it, silently; files that don't get the heading created automatically instead of asking you anything. Every file ends up with the same structure, which sounds like what you were normalizing toward anyway.

Option B, if you'd rather choose yourself in the odd-format files: enable **Choose heading when capturing**. At capture time you get a dropdown of the destination note's actual headings, and you can type one that doesn't exist yet (with Create line if not found on, QuickAdd creates it). The trade-off: it always asks, not only when `To Do` is missing.

The exact conditional you described - silent when `To Do` exists, prompt otherwise - isn't declarative in the UI. It would take a small macro user script that reads `app.metadataCache.getFileCache(file).headings` and branches. Happy to sketch that if neither option above fits your workflow.

Docs: https://quickadd.obsidian.guide/docs/Choices/CaptureChoice#insert-after
